### DISCUSSION TITLE
네이버 API 유효성 검사 기능에 대해 문의드립니다.

### DISCUSSION THREAD

**Discussion by gopherkook** [commerce-api-naver/commerce-solutions#51] [묻고 답하기]
![2024-11-08 163633](https://github.com/user-attachments/assets/25c3e1cd-4d5e-4848-883e-ee2a8fcf8554)
현재 네이버 애플리케이션 ID와 시크릿키를 연동할 때, 잘못된 정보 입력으로 인한 오류를 방지하고자 합니다.

목적:
API 입력 시 발생할 수 있는 오류를 최소화하고, 정확한 API 정보를 입력하도록 돕기 위함.

프로세스:
입력된 인증키를 확인.
잘못된 정보 입력 시: "정확한 키 값을 입력해 주세요"라는 알림을 표시.
정확한 정보 입력 시: 다음 단계로 진행.

이 프로세스를 통해 올바른 API 연동을 지원하고, 오류를 예방하고자 합니다. 따라서 해당 기능 구현을 위해 커머스 API에서 올바르지 못한 키값을 호출했을 때 유효성을 검사해주는 서비스가 있는지 여쭙습니다.

감사합니다.

**Comment by commerce-api-naver** [comment]
안녕하세요. 커머스API 이민석입니다.

## 애플리케이션 위탁은 이용약관 위배 소지가 있습니다.
판매자가 발급받는 '내 스토어 애플리케이션'은 판매자 본인이 연동/운용하는 경우에만 사용할 수 있으며 발급받은 인증 정보를 제 3자에게 위탁 제공하는 것은 이용약관 위배 소지가 있습니다.
이러한 방법은 판매자의 스토어 보안에도 문제가 있을 수 있기에 판매자 분들께는 솔루션 서비스 이용을 목적으로 사용자로 하여금 '내 스토어 애플리케이션' 발급을 유도하여 발급받은 애플리케이션 인증 정보(ID/Secret)를 요구하는 솔루션을 주의해야하며 이러한 솔루션 요구에 가급적 응하지 말것을 권장하고 있습니다.
따라서 말씀하신 기능은 이용약관 위배 소지가 있는 설계 방법으로 관련된 지원을 해드릴 수 없습니다.

## 서드파티 솔루션의 연동
판매자 분들의 스마트스토어 운용을 돕는 서드파티 솔루션을 운용중이신 경우 네이버에서 공식적으로 지원하는 연동 방법으로 연동을 진행하는 것을 권장하고 있습니다.
네이버에서는 사용자들로부터 애플리케이션 정보를 개별적으로 입력받지 않더라도 스토어 정보에 안전하게 접근할 수 있는 연동 방법을 제공하고 있습니다.
운용중이신 솔루션 성격과 형태에 따라 커머스솔루션 또는 API대행사 형태로 연동 방법을 확인해보시기 바랍니다.
- 커머스솔루션 개발 가이드: https://apicenter.commerce.naver.com/ko/basic/solution-doc
- API대행사 관련 공지사항: https://notice.naver.com/notices/cac/14592

**Reply by gopherkook** [reply]
현재 API 대행사 자격요건을 받아 운용 중입니다. API 대행사 조차 현재 설계 방법으로 관련된 지원을 받을 수 없는건가요?

**Reply by commerce-api-naver** [reply]
@gopherkook 님 안녕하세요. 커머스API 이민석입니다.

API대행사 방식은 사용자로부터 애플리케이션 인증 정보 수집 없이도 스토어 데이터 연동이 가능합니다.
API대행사 자격과는 별개로 해당하는 방식으로의 인증 정보 위탁 수집은 지원이 불가능합니다.

더불어 이 공간은 커머스솔루션 개발사를 위한 기술지원 공간입니다.
이후 API대행사 방식으로의 연동은 아래 공간으로 문의해주시기 바랍니다.
- https://github.com/commerce-api-naver/api-agency
